sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -93,7 +94,8 @@
    * No two instances of this class are considered equivalent hence we generate a random hash code
    * between 0 and {@link Integer#MAX_VALUE}.
    */
-  private final int hashCode = (int) (Math.random() * Integer.MAX_VALUE);
+  private final int hashCode = ThreadLocalRandom.current().nextInt(
```

Comment:
It turns out you ran into an undocumented limit of nextInt, it can't generate such a large range since it overflows internally (when subtracting min from max):
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/7u40-b43/java/util/concurrent/ThreadLocalRandom.java?av=f#143

I'll merge and update the bound to use 0 as the minimum